### ISSUE TITLE
This requires a bit of a jump

### ISSUE THREAD

**Issue by davehodg** [etadata/owasp-workshop#1]
"Create 3 hosts and call them node-master, node-1 and node-2."

Needs a bit of explanation!

If I do:

$ docker run --name node-master -i -t ubuntu /bin/bash

And then do your suggested apt stuff, the kubeadm apt isn't found.

$ docker run  -i -t ubuntu:16.04 /bin/bash

It was a hell of a job getting kubadm installed. Your stanza didn't work needed to add "deb http://apt.kubernetes.io/ kubernetes-xenial main" manually.

kubeadm init needed --ignore-preflight-errors=all and even then, ended with a load of:

	[WARNING ImagePull]: failed to pull image k8s.gcr.io/kube-apiserver:v1.12.2: output: Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
, error: exit status 1

The docker daemon isn't running despite installing docker.

**Comment by etadata** [comment]
for the part 1 of the lab,  you can download a preconfigured machine from [here](https://public.boxcloud.com/d/1/b1!04Ax6AGgLnW4f45JYQ0jo8Pm7gCfuFaLkUJ4zw2GOPQ4d-k3kxzOtnqAWlg3MLt1xjzcp5HCSNVR4YAo1AT8mVI1XQC85giOJ__9SQZc3PN0zcDljIhdXT7EsTV3s-nGoIdfuaKw2gVfbyuLqhbHOz6IK7LADt-Jrn_eIG0qA_5VXrdQSDmEglw-58Wt--IFuI0m06cqfXke1ms54_uBvXC6uSOmoywxj04eD2Fqut-TOnFBCE7YQvc8QmSwTufbCH0iZZzlyBH7mtdMMnmWc6Mh5r7avAJH-4SoNKVZRMVO5f-dYtSLSB8H8thYEuhueEFts6d32ypYTlbm4lnG-7guIFoaYlkKkHp2GeZu54jo8DDLj6p8JU0ehgJI3CKOPaasOvzppBtmz4W43VJgOeOHr9AdQu5ALzcuqRS0lA8wgNyBn7yX45YRHzZ7cc1zV0yGge1JKhAKvSIXg9XZQG5wMZennOLucpH2ObhOvjuhsQm8w3mFmhXC8orIBDjSqUacxIcvttWBvgL46YvlJ0ahIFz7uKWMoIhoWkvG5FhMSOrWcTaPMkwNn6-lX4QKcMcF5X_T8SgiIOxEboKT7VqRMeUXrKMqo0IWQbAqO7eN-ekeu1QNr4wk8-liH-bDcIwf4P7VxwBAO_sLRxzxSDSzOWMOevGzfEUflRl6q3G6blKypjBBTtBSB4CBk-f7dVhhjPPsyWiIwEsiMgMQLVtm9hX-4ac373tU1KXfH2EqGe2i_VWAnBQ1WC8Z1I3xHlL6mkG1x0aRkjtrFxlwhhQ9WUAmlNFXgkn7-cXSfC9RFoaA_oCcHtxYbzBmz8xNFtO4BCKY9LYzD9SPNGgFoZN-wQEft-4w_tVJjwRJuOl2TqfX1YNgd3-37a283wJs-dCwosl63geBDvkJFumdJRo8R0G3qlZZk7HfInTAd4mubfBgMtAo4NjkropYM4-UPhc0j5WwEbyZcyy2PIvGAmH71piy7JSdD8C-xI3GlAfeg6ovuAMsfHmx0tqG0UlvcwRZyqWBYpN18BYFIpbiSVu3eH-yIzKG39D4NebIXD3MlUy9GOmz09m7Hg-xfVDZ-kDWjvOC2Vp8FOXVowSooO5gVb0s6H0i_TDyiDxua_0B9o2bLiaHLSWQui-fIKCi2mYS1YvVn-8ziJb3VKbxWtoaC9Iv6zfQw4FTi9K_rbTYvKh5z8AIJGNDOO4R31l5pRm3xS8ggqDM6k-HaQqRVLQKt_OtQUOI5A../download)

according to the error message, it seems like the docker service isn't running 

inside the node-1 and node-2 
```
for SERVICES in kube-proxy kubelet  docker; do
    systemctl restart $SERVICES
    systemctl enable $SERVICES
    systemctl status $SERVICES
done
```
to see if the docker daemon is already running 
`
systemctl status docker
`
`docker info`

to restart the services in the node-master
```
for SERVICES in etcd kube-apiserver kube-controller-manager kube-scheduler flanneld; do
	systemctl restart $SERVICES
	systemctl enable $SERVICES
	systemctl status $SERVICES
done
```
and plz make sure the DNS service is already configured
another tip: if you are running the cluster's machine has more than one NIC, then k8s will use the one which has the default route 

kubeadm must be run `from` the node-master

**Comment by davehodg** [comment]
Why the nodes? GitHub says to do the kube stuff on the master.

Trying your image now. Oh, link didn't work.

**Comment by etadata** [comment]
exactly, the kube staff should be on the master. and you have to administrate the nodes using kubectl from the master.
i didn't upload a preconfigured VM. i used the GCP to create k8s cluster.
if you would like to create k8s cluster by yourself try to download this [ubuntu iso](http://releases.ubuntu.com/16.04/ubuntu-16.04.5-server-amd64.iso)
and follow the steps in the option.

I will be glad to help you create ur cluster
